litmus-portal/frontend/src/pages/MonitoringDashboardPage/index.tsx

```diff
@@ -98,70 +98,69 @@ const DashboardPage: React.FC = () => {
-
-  useEffect(() => {
-    if (selectedDataSource.selectedDataSourceID === '') {
-      dashboard.selectDashboard({
-        refreshRate,
-      });
+    if (dashboards && dashboards.ListDashboard.length) {
       if (
-        dataSources &&
-        selectedDashboardInformation.metaData &&
-        selectedDashboardInformation.metaData[0] &&
-        dataSources.ListDataSource
+        selectedDashboardInformation.id !==
+        selectedDashboardInformation.dashboardKey
       ) {
-        const selectedDataSource: ListDataSourceResponse = dataSources.ListDataSource.filter(
+        const availableDashboards: ListDashboardResponse[] = dashboards.ListDashboard.filter(
           (data) => {
-            return (
-              data.ds_id === selectedDashboardInformation.metaData[0].ds_id
-            );
+            return data.cluster_id === selectedDashboardInformation.agentID;
+          }
+        );
+        const selectedDashboard: ListDashboardResponse = availableDashboards.filter(
+          (data) => {
+            return data.db_id === selectedDashboardInformation.id;
           }
         )[0];
-        if (selectedDataSource) {
-          dataSource.selectDataSource({
-            selectedDataSourceURL: selectedDataSource.ds_url,
-            selectedDataSourceID: selectedDataSource.ds_id,
-            selectedDataSourceName: selectedDataSource.ds_name,
-          });
-        }
+        setSelectedDashboardInformation({
+          ...selectedDashboardInformation,
+          dashboardListForAgent: availableDashboards,
+          metaData: [selectedDashboard],
+          dashboardKey: selectedDashboardInformation.id,
+        });
+      }
+    }
+  }, [dashboards, selectedDashboardInformation.id]);
+
+  useEffect(() => {
+    if (dataSources && dataSources.ListDataSource.length) {
+      if (selectedDataSource.selectedDataSourceID === '') {
+        dashboard.selectDashboard({
+          refreshRate,
+        });
         if (
-          selectedDataSource &&
-          selectedDataSource.health_status !== 'Active'
+          selectedDashboardInformation.metaData &&
+          selectedDashboardInformation.metaData[0] &&
+          dataSources.ListDataSource
         ) {
-          setDataSourceStatus(selectedDataSource.health_status);
+          const selectedDataSource: ListDataSourceResponse = dataSources.ListDataSource.filter(
+            (data) => {
+              return (
+                data.ds_id === selectedDashboardInformation.metaData[0].ds_id
+              );
+            }
+          )[0];
+          if (selectedDataSource) {
+            dataSource.selectDataSource({
+              selectedDataSourceURL: selectedDataSource.ds_url,
+              selectedDataSourceID: selectedDataSource.ds_id,
+              selectedDataSourceName: selectedDataSource.ds_name,
+            });
+          }
+          if (
+            selectedDataSource &&
+            selectedDataSource.health_status !== 'Active'
```

Comment:
Can you use constants instead of strings.

Reply:
Sure I am updating this.